File: frontend/src/types/restaurant.ts
/**
 * Restaurant-related type definitions
 */

export interface Restaurant {
  id: string;
  name: string;
  rating?: number;
  review_count?: number;
  price?: string;
  categories?: Category[];
  location?: RestaurantLocation;
  phone?: string;
  image_url?: string;
  url?: string;
  is_closed?: boolean;
  distance?: number;
  reviews?: Review[];
}

export interface Category {
  alias: string;
  title: string;
}

export interface RestaurantLocation {
  address1?: string;
  address2?: string;
  address3?: string;
  city?: string;
  state?: string;
  zip_code?: string;
  country?: string;
  display_address?: string[];
}

export interface Review {
  id: string;
  rating: number;
  text: string;
  time_created: string;
  user?: {
    name?: string;
    image_url?: string;
  };
}

export interface RestaurantSearchParams {
  query?: string;
  location?: string;
  latitude?: number;
  longitude?: number;
  categories?: string;
  price?: string;
  radius?: number;
  limit?: number;
  sort_by?: string;
}

File: backend/app/models/restaurant.py
"""Restaurant data models"""

from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field


class Location(BaseModel):
    """Location model"""
    latitude: float
    longitude: float
    address: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    zip_code: Optional[str] = None
    country: Optional[str] = None


class Category(BaseModel):
    """Restaurant category"""
    alias: str
    title: str


class Review(BaseModel):
    """Restaurant review"""
    id: str
    rating: float
    text: str
    time_created: str
    user_name: Optional[str] = None


class Restaurant(BaseModel):
    """Restaurant model"""
    id: str
    name: str
    rating: Optional[float] = None
    review_count: Optional[int] = 0
    price: Optional[str] = None
    categories: List[Category] = Field(default_factory=list)
    location: Optional[Location] = None
    phone: Optional[str] = None
    image_url: Optional[str] = None
    url: Optional[str] = None
    is_closed: bool = False
    distance: Optional[float] = None
    reviews: Optional[List[Review]] = Field(default_factory=list)
    menu_summary: Optional[str] = None


class RestaurantSearchParams(BaseModel):
    """Search parameters for restaurants"""
    location: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    term: Optional[str] = None
    categories: Optional[str] = None
    price: Optional[str] = None  # "1,2,3,4"
    radius: int = 5000  # meters
    limit: int = 20
    sort_by: Optional[str] = "best_match"  # best_match, rating, review_count, distance
    open_now: bool = False

